React/class-components.md

```diff
@@ -21,9 +21,19 @@ const Student = (props) => {
 };
 ```
 
-These are called, _functional components,_ because they are composed of one function.  Functional components take in data from `props` and return JSX.  These components are easy to read, test and debug because given a set of props they always return the same JSX back.  However often we want to keep track of information over time, or the _state_ of our component.  This is why functional components are often called _Stateless Components_ in React.  They take props as input and return JSX, which can be a mixture of nested components and HTML.
+These are called, _functional components,_ because they are composed of one function.  Functional components take in data from `props` and return JSX.  When functional components **only** use props and not the `useState` hook these functional components are often called _Stateless Components_ in React.  These stateless components are attractive because they are relatively straightforward to read, test and debug.  
 
-If you remember back to Ruby, we built objects which combined data, which could change over time, with functionality by defining classes.  We can build React components which can do the same!
+However we often need to keep track of information over time or the _state_ of our component like we have done with the `useState` hook.
+
+If you remember back to Ruby, we built objects which combined data, which could change over time, with functionality by defining classes.  React _class components_ provide an alternative to the `useState` hook to manage the state of a component, and provide a set of lifecycle methods to manage a component's rendering and state.
+
+### So Why Didn't We Learn Class Components Earlier?  
+
+At Ada we feel that, in most cases, that the recent introduction of hooks make class components _unnecessary_.  With hooks most applications can be written exclusively with functional components.  There are a few [specialized cases](https://reactjs.org/docs/error-boundaries.html) which require class components, but these scenarios are few and far between and may be replaced hook-based solutions at a later date.
+
+### So Why Learn About Class Components?
+
+React has been around for a while now and a great deal of code and documentation has been written using class components.  React is not eliminating class components and a great number of developers find the abstraction of class components attractive.  It's important to know how they work and how to convert between functional and class components.
```

Comment:
This is all really helpful context and good reasons to take the time to learn class components. 
That said, I wonder if it would be worth pulling this out of the curriculum, at least for C13 because:
- We are expecting to move at a little slower pace (from my understanding), so considering we are adding content about the useState hook I expect we'll need to cut something.
- I would argue that since our goal is to prepare them to be full-stack devs, not so much to be react devs specifically, we could expect that any students who need to can learn about class components on the job and many students will never need to.

Reply:
We can definitely make this section optional.  I'll bring this up at our next standup.

💯

Reply:
I really like making this an optional reading.